src/main/java/com/google/sps/servlets/PhotosServlet.java

```diff
@@ -0,0 +1,167 @@
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.BlurImage;
+import com.google.sps.data.LoggedUser;
+import com.google.sps.data.User;
+import java.awt.Point;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/photos")
+public class PhotosServlet extends HttpServlet {
+
+  /** 
+   * Method that handles the GET requests to "/photos" path. Parameter "max-photos" specifies the
+   * maximum number of photos to return. Returns a JSON array of BlurImages ordered by dateAccessed
+   * descending.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // If the user is not logged in, send a redirect to the main page.
+    User user = User.getCurrentUser();
+    if (!user.isLoggedIn()) {
+      response.sendRedirect("/");
+      return;
+    }
+    LoggedUser loggedUser = (LoggedUser) user;
+
+    // Get the input from the request.
+    String maxPhotosString = request.getParameter("max-photos");
+
+    // Convert the input to an int or Integer.MAX_VALUE.
+    int maxPhotos;
+    try {
+      maxPhotos = Integer.parseInt(maxPhotosString);
+    } catch (NumberFormatException e) {
+      maxPhotos = Integer.MAX_VALUE;
+    }
+
+    // Make sure maxPhotos is not negative.
+    if(maxPhotos < 0) {
+        maxPhotos = 0;
+    }
+
+    // Load current user's photos from datastore.
+    String userId = loggedUser.getId();
+    Query query = new Query("BlurImage");
+    query.setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, userId));
+    query.addSort("dateAccessed", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Store photos in an array.
+    ArrayList<BlurImage> photos = new ArrayList<>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(maxPhotos))) {
+      long id = entity.getKey().getId();
+      BlobKey blobKey = (BlobKey) entity.getProperty("blobKey");
+      List<List<Point>> blurRectangles = (List<List<Point>>) entity.getProperty("blurRectangles");
+      Date dateAccessed = (Date) entity.getProperty("dateAccessed");
```

Comment:
Probably update the dateAccessed property or call it dateCreated

Reply:
Good point! Thank you.